eva-server/src/main/java/uk/ac/ebi/eva/server/ws/VariantWSServerV2.java

```diff
@@ -0,0 +1,227 @@
+/*
+ * European Variation Archive (EVA) - Open-access database of all types of genetic
+ * variation data from all species
+ *
+ * Copyright 2019 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.ac.ebi.eva.server.ws;
+
+import io.swagger.annotations.Api;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.hateoas.Link;
+import org.springframework.hateoas.Resource;
+import uk.ac.ebi.eva.commons.core.models.Annotation;
+import uk.ac.ebi.eva.commons.core.models.AnnotationMetadata;
+import uk.ac.ebi.eva.commons.core.models.ws.VariantSourceEntryWithSampleNames;
+import uk.ac.ebi.eva.commons.core.models.ws.VariantWithSamplesAndAnnotation;
+import uk.ac.ebi.eva.commons.mongodb.services.AnnotationMetadataNotFoundException;
+import uk.ac.ebi.eva.commons.mongodb.services.VariantWithSamplesAndAnnotationsService;
+import uk.ac.ebi.eva.lib.eva_utils.DBAdaptorConnector;
+import uk.ac.ebi.eva.lib.eva_utils.MultiMongoDbFactory;
+import uk.ac.ebi.eva.lib.utils.QueryResponse;
+import uk.ac.ebi.eva.lib.utils.QueryResult;
+
+import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.springframework.hateoas.mvc.ControllerLinkBuilder.linkTo;
+import static org.springframework.hateoas.mvc.ControllerLinkBuilder.methodOn;
+
+@RestController
+@RequestMapping(value = "/v2/variants", produces = "application/json")
+@Api(tags = {"variants"})
+public class VariantWSServerV2 extends EvaWSServer {
+
+    @Autowired
+    private VariantWithSamplesAndAnnotationsService service;
+
+    @GetMapping(value = "/{variantId}")
+    public Resource<QueryResponse> getCoreInfo(@PathVariable("variantId") String variantId,
+                                               @RequestParam(name = "species") String species,
+                                               @RequestParam(name = "annot-vep-version", required = false)
```

Comment:
I am sorry, I didn't understand this.
Could you please explain? :)

Reply:
What I mean is when you search for a particular variant `1:1000:A:C`, the version of the annotation is not important for retrieving the core variant information.
The only reason why we could need the related parameters (`annot-vep-version` and `annot-vep-cache-version`) would be to construct the HATEOAS links to the same annotation as requested, instead of displaying them as generic ones. But I think it is preferrable to show them as generic, and including a separate endpoint that returns which versions are available.

Is this of any help? :)

Reply:
Yeah :+1: 
Fixed :)